salt/modules/kubernetes.py

```diff
@@ -64,16 +80,31 @@ def _setup_conn(**kwargs):
                                      'http://localhost:8080')
     username = __salt__['config.option']('kubernetes.user')
     password = __salt__['config.option']('kubernetes.password')
+    ca_cert = __salt__['config.option']('kubernetes.certificate-authority-data')
+    client_cert = __salt__['config.option']('kubernetes.client-certificate-data')
+    client_key = __salt__['config.option']('kubernetes.client-key-data')
+    ca_cert_file = __salt__['config.option']('kubernetes.certificate-authority-file')
+    client_cert_file = __salt__['config.option']('kubernetes.client-certificate-file')
+    client_key_file = __salt__['config.option']('kubernetes.client-key-file')
 
     # Override default API settings when settings are provided
-    if kwargs.get('api_url'):
-        host = kwargs['api_url']
+    if 'api_url' in kwargs:
```

Comment:
@cachedout - To make it clear. The current code look like this:

```
host = __salt__['config.option']('kubernetes.api_url',
                                 'http://localhost:8080')
if 'api_url' in kwargs:
    host = kwargs.get('api_url')
```

Using just `get()` would overwrite the value got from the config

```
host = __salt__['config.option']('kubernetes.api_url',
￼                                       'http://localhost:8080')
host = kwargs.get('api_url')
```
If api_url is not defined in kwargs (which should be the case in 99% of the usage of this API) `host` will always become None.